modulemd/v2/modulemd-profile.c

```diff
@@ -155,7 +160,17 @@ modulemd_profile_get_description (ModulemdProfile *self, const gchar *locale)
 {
   g_return_val_if_fail (MODULEMD_IS_PROFILE (self), NULL);
 
-  /* TODO: retrieve translated strings */
+  /* Retrieve translated strings, if available */
+  if (self->owner)
+    {
+      ModulemdTranslationEntry *entry =
+        modulemd_module_stream_get_translation_entry (self->owner, locale);
+
+      if (entry != NULL && modulemd_translation_entry_get_profile_description (
+                             entry, self->name) != NULL)
+        return modulemd_translation_entry_get_profile_description (entry,
+                                                                   self->name);
```

Comment:
(Optional) The above would be a little easier to grok if the flow was along the lines of
```
if (entry != NULL)
  {
    gchar *translation = modulemd_translation_entry_get_profile_description (
                             entry, self->name);
    if (translation != NULL)
      return translation;
  }
```